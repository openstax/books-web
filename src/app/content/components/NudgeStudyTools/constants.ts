export const arrowDesktopHeight = 13;
export const arrowMobileHeight = 6.2;
export const arrowLeftMargin = 1;
export const arrowTopMargin = 1;

export const closeButtonDistanceFromContent = 4;
export const closeButtonMobileMargin = 3;

export const contentMarginTop = 1;

export const spotlightPadding = 1;

export const cookieNudge = {
  // How many times nuge was shown
  counter: 'nudge_study_guides_counter',
  // When nudge was shown
  date: 'nudge_study_guides_date',
  // How many pages user has opened (opening the same page twice counts it as 2)
  pageCounter: 'nudge_study_guides_page_counter',
};

// Show nudge only for users which opened at least 2 pages
export const nudgeStudyToolsMinPageLimit = 2;

// Show nudge every 4 weeks: 4 * 7 days * 24 hours * 60 minutes * 60 seconds * 1000 ms
export const timeIntervalBetweenShowingNudgeInMs = 4 * 7 * 24 * 60 * 60 * 1000;
// Show nudge max 3 times
export const nudgeStudyToolsShowLimit = 3;
// Id of the element which should be highlighted
export const nudgeStudyToolsTargetId = 'nudge-study-tools';
