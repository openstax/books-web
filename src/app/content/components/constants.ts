import theme from '../../theme';

export const bookBannerDesktopBigHeight = 13;
export const bookBannerDesktopMiniHeight = 7;
export const bookBannerMobileBigHeight = 10.4;
export const bookBannerMobileMiniHeight = 6;

export const sidebarDesktopWidth = 33.5;
export const sidebarMobileWidth = 28.8;
export const sidebarTransitionTime = 300;

export const toolbarIconColor = theme.color.primary.gray;
export const toolbarDesktopHeight = 5;
export const toolbarMobileHeight = 4;

export const contentTextWidth = 84;

export const mainContentBackground = '#fff';

export const maxContentGutter = 6;
export const contentWrapperMaxWidth = contentTextWidth + sidebarDesktopWidth + maxContentGutter * 2;
