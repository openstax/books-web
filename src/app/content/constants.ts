import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import theme from '../theme';

export const studyGuidesFeatureFlag = 'studyGuidesEnabled';
export const practiceQuestionsFeatureFlag = 'practiceQuestionsEnabled';

export const maxHighlightsApiPageSize = 200;
export const maxResourcesPerFetch = 10;

export const summaryPageSize = 20;
export const loadMoreDistanceFromBottom = 50;

export const modalQueryParameterName = 'modal';

export const highlightStyles: Array<{label: HighlightColorEnum, passive: string, focused: string}> = [
  {label: HighlightColorEnum.Yellow, passive: '#ffff8a', focused: '#fed200'},
  {label: HighlightColorEnum.Green, passive: '#def99f', focused: '#92d101'},
  {label: HighlightColorEnum.Blue, passive: '#c8f5ff', focused: '#00c3ed'},
  {label: HighlightColorEnum.Purple, passive: '#cbcfff', focused: '#545ec8'},
  {label: HighlightColorEnum.Pink, passive: '#ffc5e1', focused: '#de017e'},
];

const getPrimaryBaseColors = (primaryColors: {[key: string]: {[key: string]: string}}) => {
  const primaryBases: { [key: string]: string } = {};

  // tslint:disable-next-line: max-line-length
  Object.keys(primaryColors).forEach((color) => {primaryBases[color] = primaryColors[color].base; });
  return primaryBases;
};

export const searchButtonStyles: {[key: string]: string} = {
  gray: '#5D6062',
  transparent: 'transparent',
  ...getPrimaryBaseColors(theme.color.primary) as {},
};
