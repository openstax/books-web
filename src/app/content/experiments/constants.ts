export const experiments: { [key: string]: any } = {
  searchButton: [
    'originalButtonEnabled',
    'bannerColorButtonEnabled',
    'grayButtonEnabled',
  ],
};

export const experimentIds: { [key: string]: any } = {
  OCCkMMCZSwW87szzpniCow: 'searchButton',
};
