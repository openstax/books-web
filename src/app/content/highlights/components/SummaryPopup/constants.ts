export const mobilePaddingSides = 1.6;
export const mobilePaddingTopBottom = 1.4;
export const mobileMarginSides = 0.8;
export const mobileMarginTopBottom = 2;
