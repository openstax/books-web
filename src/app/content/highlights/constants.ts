import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import books from '../../../config.books';

export const maxHighlightsApiPageSize = 200;
export const maxResourcesPerFetch = 10;

export const summaryPageSize = 20;
export const loadMoreDistanceFromBottom = 50;

export const cardWidth = 20;
export const cardPadding = 0.8;
export const cardContentMargin = 3;
export const cardFocusedContentMargin = 1;
export const cardMinWindowMargin = 2;
export const cardMarginBottom = 2;

export const highlightStyles: Array<{label: HighlightColorEnum, passive: string, focused: string}> = [
  {label: HighlightColorEnum.Yellow, passive: '#ffff8a', focused: '#fed200'},
  {label: HighlightColorEnum.Green, passive: '#def99f', focused: '#92d101'},
  {label: HighlightColorEnum.Blue, passive: '#c8f5ff', focused: '#00c3ed'},
  {label: HighlightColorEnum.Purple, passive: '#cbcfff', focused: '#545ec8'},
  {label: HighlightColorEnum.Pink, passive: '#ffc5e1', focused: '#de017e'},
];

export const enabledForBooks = Object.keys(books);

export const highlightBlockPadding = 1;
export const highlightIndicatorSizeForBlock = 1.2;
export const highlightIndicatorSize = 0.9;
