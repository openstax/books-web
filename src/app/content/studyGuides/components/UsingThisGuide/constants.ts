export const cookieUTG = 'user_has_seen_UTG';
