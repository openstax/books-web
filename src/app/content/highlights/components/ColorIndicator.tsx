import React from 'react';
import styled, { css } from 'styled-components/macro';
import { Check } from 'styled-icons/fa-solid/Check';
import { isDefined } from '../../../guards';
import { highlightStyles } from '../constants';

interface Props<T extends React.ComponentType | undefined = React.ComponentType> {
  size?: 'small';
  shape?: 'square' | 'circle';
  className?: string;
  style: typeof highlightStyles[number];
  checked?: boolean;
  component?: T extends undefined ? undefined :
    T extends React.ComponentType ? React.ReactComponentElement<T>:
    never;
}

// tslint:disable-next-line:variable-name
const CheckIcon = styled(Check)`
  display: none;
`;

// tslint:disable-next-line:variable-name
function Hoc<T extends React.ComponentType | undefined>(
  {children, style, checked, size, component, ...props}: React.PropsWithChildren<Props<T>>
) {
  if (isDefined(component)) {
    return React.cloneElement(component, props, [children, <CheckIcon key='check' />]);
  }
  return <div {...props}><CheckIcon />{children}</div>;
}

const indicatorSize = (props: {size?: 'small'}) => props.size === 'small' ? 1.6 : 2.4;
const checkSize = (props: {size?: 'small'}) => props.size === 'small' ? 1 : 1.6;

// tslint:disable-next-line:variable-name
const ColorIndicator = styled(Hoc)`
  background-color: ${(props: {style: typeof highlightStyles[number]}) => props.style.passive};
  border: 1px solid ${(props: {style: typeof highlightStyles[number]}) => props.style.focused};
  height: ${(props: Props) => indicatorSize(props)}rem;
  width: ${(props: Props) => indicatorSize(props)}rem;
  display: flex;
  justify-content: center;
  align-items: center;
  ${(props: Props) => (props.shape === 'circle' || props.shape === undefined) && css`
    border-radius: 2rem;
  `}

  ${CheckIcon} {
    height: ${(props: Props) => checkSize(props)}rem;
    width: ${(props: Props) => checkSize(props)}rem;
    ${(props: {checked?: boolean, style: typeof highlightStyles[number]}) => props.checked && css`
      color: ${props.style.focused};
      display: block;
    `}
  }

  ${(props: {checked?: boolean, style: typeof highlightStyles[number]}) => props.checked !== false && css`
    input:checked + ${CheckIcon} {
      color: ${props.style.focused};
      display: block;
    }
  `}
`;

export default ColorIndicator;
