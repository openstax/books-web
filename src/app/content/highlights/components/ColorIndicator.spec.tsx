import React from 'react';
import renderer from 'react-test-renderer';
import { highlightStyles } from '../../constants';
import ColorIndicator from './ColorIndicator';

describe('ColorIndicator', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<ColorIndicator style={highlightStyles[0]} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot (checked)', () => {
    const component = renderer.create(<ColorIndicator checked style={highlightStyles[0]} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
