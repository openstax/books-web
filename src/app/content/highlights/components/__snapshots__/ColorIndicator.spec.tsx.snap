// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorIndicator matches snapshot (checked) 1`] = `
.c3 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c2 {
  display: none;
}

.c4 .c1 {
  height: 1.6rem;
  width: 1.6rem;
}

.c4 input:checked + .c1 {
  color: #fed200;
  display: block;
}

.c0 {
  background-color: #ffff8a;
  border: 1px solid #fed200;
  height: 2.4rem;
  width: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2rem;
}

.c0 .c1 {
  height: 1.6rem;
  width: 1.6rem;
  color: #fed200;
  display: block;
}

.c0 input:checked + .c1 {
  color: #fed200;
  display: block;
}

<div
  className="c0"
>
  <svg
    aria-hidden="true"
    className="c1 c2 c3"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 512 512"
  >
    <path
      d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`ColorIndicator matches snapshot 1`] = `
.c3 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c2 {
  display: none;
}

.c0 {
  background-color: #ffff8a;
  border: 1px solid #fed200;
  height: 2.4rem;
  width: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2rem;
}

.c0 .c1 {
  height: 1.6rem;
  width: 1.6rem;
}

.c0 input:checked + .c1 {
  color: #fed200;
  display: block;
}

<div
  className="c0"
>
  <svg
    aria-hidden="true"
    className="c1 c2 c3"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 512 512"
  >
    <path
      d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
      fill="currentColor"
    />
  </svg>
</div>
`;
